### PR TITLE
Use `REPORT_OUTPUT_FOLDER: none` to not generate report files

### PR DESCRIPTION
**Pull request by nvuillam**
Fixes https://github.com/oxsecurity/megalinter/issues/1638

### CODE REVIEW

**Comment by nvuillam** [comment]
## [MegaLinter](https://oxsecurity.github.io/megalinter/latest) status: :warning: [**WARNING**](https://github.com/oxsecurity/megalinter/actions/runs/2767565340)

|    Descriptor    |                                                         Linter                                                         |Files|Fixed|                                  Errors                                  |Elapsed time|
|------------------|------------------------------------------------------------------------------------------------------------------------|-----|----:|--------------------------------------------------------------------------|------------|
|✅ BASH           |[bash-exec](https://oxsecurity.github.io/megalinter/latest/descriptors/bash_bash_exec)                                  |    7|     |                                                                         0|0.02s       |
|✅ BASH           |[shellcheck](https://oxsecurity.github.io/megalinter/latest/descriptors/bash_shellcheck)                                |    7|     |                                                                         0|0.25s       |
|✅ BASH           |[shfmt](https://oxsecurity.github.io/megalinter/latest/descriptors/bash_shfmt)                                          |    7|    0|                                                                         0|0.28s       |
|✅ COPYPASTE      |[jscpd](https://oxsecurity.github.io/megalinter/latest/descriptors/copypaste_jscpd)                                     |yes  |     |no                                                                        |1.85s       |
|✅ DOCKERFILE     |[hadolint](https://oxsecurity.github.io/megalinter/latest/descriptors/dockerfile_hadolint)                              |  102|     |                                                                         0|18.29s      |
|✅ JSON           |[eslint-plugin-jsonc](https://oxsecurity.github.io/megalinter/latest/descriptors/json_eslint_plugin_jsonc)              |   20|    0|                                                                         0|1.96s       |
|✅ JSON           |[jsonlint](https://oxsecurity.github.io/megalinter/latest/descriptors/json_jsonlint)                                    |   18|     |                                                                         0|3.66s       |
|✅ JSON           |[v8r](https://oxsecurity.github.io/megalinter/latest/descriptors/json_v8r)                                              |   20|     |                                                                         0|10.62s      |
|:warning: MARKDOWN|[markdownlint](https://oxsecurity.github.io/megalinter/latest/descriptors/markdown_markdownlint)                        |  279|    0|[**4**](https://github.com/oxsecurity/megalinter/actions/runs/2767565340) |4.37s       |
|✅ MARKDOWN       |[markdown-link-check](https://oxsecurity.github.io/megalinter/latest/descriptors/markdown_markdown_link_check)          |  279|     |                                                                         0|4.71s       |
|✅ MARKDOWN       |[markdown-table-formatter](https://oxsecurity.github.io/megalinter/latest/descriptors/markdown_markdown_table_formatter)|  279|    0|                                                                         0|6.4s        |
|✅ OPENAPI        |[spectral](https://oxsecurity.github.io/megalinter/latest/descriptors/openapi_spectral)                                 |    1|     |                                                                         0|0.8s        |
|:warning: PYTHON  |[bandit](https://oxsecurity.github.io/megalinter/latest/descriptors/python_bandit)                                      |  163|     |[**50**](https://github.com/oxsecurity/megalinter/actions/runs/2767565340)|1.99s       |
|✅ PYTHON         |[black](https://oxsecurity.github.io/megalinter/latest/descriptors/python_black)                                        |  163|    0|                                                                         0|3.03s       |
|✅ PYTHON         |[flake8](https://oxsecurity.github.io/megalinter/latest/descriptors/python_flake8)                                      |  163|     |                                                                         0|1.73s       |
|✅ PYTHON         |[isort](https://oxsecurity.github.io/megalinter/latest/descriptors/python_isort)                                        |  163|    0|                                                                         0|0.56s       |
|✅ PYTHON         |[mypy](https://oxsecurity.github.io/megalinter/latest/descriptors/python_mypy)                                          |  163|     |                                                                         0|87.4s       |
|✅ PYTHON         |[pylint](https://oxsecurity.github.io/megalinter/latest/descriptors/python_pylint)                                      |  163|     |                                                                         0|14.25s      |
|✅ REPOSITORY     |[git_diff](https://oxsecurity.github.io/megalinter/latest/descriptors/repository_git_diff)                              |yes  |     |no                                                                        |0.27s       |
|✅ REPOSITORY     |[secretlint](https://oxsecurity.github.io/megalinter/latest/descriptors/repository_secretlint)                          |yes  |     |no                                                                        |11.4s       |
|✅ REPOSITORY     |[trivy](https://oxsecurity.github.io/megalinter/latest/descriptors/repository_trivy)                                    |yes  |     |no                                                                        |20.55s      |
|✅ SPELL          |[cspell](https://oxsecurity.github.io/megalinter/latest/descriptors/spell_cspell)                                       |  656|     |                                                                         0|18.46s      |
|✅ SPELL          |[misspell](https://oxsecurity.github.io/megalinter/latest/descriptors/spell_misspell)                                   |  488|    0|                                                                         0|0.67s       |
|✅ XML            |[xmllint](https://oxsecurity.github.io/megalinter/latest/descriptors/xml_xmllint)                                       |    3|     |                                                                         0|0.0s        |
|✅ YAML           |[prettier](https://oxsecurity.github.io/megalinter/latest/descriptors/yaml_prettier)                                    |   79|    0|                                                                         0|1.96s       |
|✅ YAML           |[v8r](https://oxsecurity.github.io/megalinter/latest/descriptors/yaml_v8r)                                              |   22|     |                                                                         0|24.96s      |
|✅ YAML           |[yamllint](https://oxsecurity.github.io/megalinter/latest/descriptors/yaml_yamllint)                                    |   80|     |                                                                         0|1.06s       |

See errors details in [**artifact MegaLinter reports** on CI Job page](https://github.com/oxsecurity/megalinter/actions/runs/2767565340)

File: README.md [hunk 1]
```diff
@@ -695,7 +695,7 @@ Configuration is assisted with auto-completion and validation in most commonly u
...
 | **PRINT_ALL_FILES**                                        | `false`                                  | Display all files analyzed by the linter instead of only the number                                                                                                                                        |
-| **REPORT_OUTPUT_FOLDER**                                   | `${GITHUB_WORKSPACE}/megalinter-reports` | Directory for generating report files                                                                                                                                                                      |
+| **REPORT_OUTPUT_FOLDER**                                   | `${GITHUB_WORKSPACE}/megalinter-reports` | Directory for generating report files. Send `none` to not generate reports                                                                                                                                 |
```
Comment (bittner) [inline README.md:698]: Typo. This should probably say:

> Set to `none` to not generate reports

File: megalinter/MegaLinter.py [hunk 2]
```diff
@@ -684,6 +684,9 @@ def initialize_output(self):
             elif os.path.isdir(self.arg_output):
                 # --output /logs/megalinter
                 self.report_folder = self.arg_output
+        # Do not initialize reports if report folder is none or false
+        if self.report_folder == "none" or self.report_folder == "false":
+            return
```
Comment (bittner) [inline megalinter/MegaLinter.py:689]: For some reason this code has no effect in the latest v6 image on DockerHub.

Setting `REPORT_OUTPUT_FOLDER: none` in the `.mega-linter.yml` config file creates a reports folder called `none` (owned by `root`).

Comment (nvuillam) [reply megalinter/MegaLinter.py:689]: Plz can u post a new issue so we don't forget to solve it ? :)

Comment (bittner) [reply megalinter/MegaLinter.py:689]: Done. #2108

**Pull request merged**